scripts/install/sonarrv3.sh

```diff
@@ -0,0 +1,179 @@
+        # Instead we symlink so postinst knows where it's at.
+        if [ -f "/usr/lib/sonarr/nzbdrone-appdata" ]; then
+            rm "/usr/lib/sonarr/nzbdrone-appdata"
+        else
+            mkdir -p "/usr/lib/sonarr"
+        fi
+        ln -s /home/"${sonarrv2owner}"/.config/NzbDrone /usr/lib/sonarr/nzbdrone-appdata
+        # chown -R "$master":"$master" /usr/lib/sonarr/nzbdrone-appdata
+
+        echo "Removing Sonarr v2" | tee -a $log
+        # shellcheck source=scripts/remove/sonarr.sh
+        bash /etc/swizzin/scripts/remove/sonarr.sh
+    fi
+}
+
+_add_sonarr_repos () {
+    echo "Adding apt sources for Sonarr v3" | tee -a $log
+    codename=$(lsb_release -cs)
+    distribution=$(lsb_release -is)
+
+    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 2009837CBFFD68F45BC180471F4F90DE2A9B4BF8 >> $log 2>&1
+    echo "deb https://apt.sonarr.tv/${distribution,,} ${codename,,} main" | tee /etc/apt/sources.list.d/sonarr.list >> $log 2>&1
+
+    #shellcheck source=sources/functions/mono
+    . /etc/swizzin/sources/functions/mono
+    mono_repo_setup
+
+    apt_update
+
+    if ! apt-cache policy sonarr | grep -q apt.sonarr.tv ; then
+        echo "Sonarr was not found from apt.sonarr.tv repository. Please inspect the logs and try again later."
+        exit 1
+    fi
+
+}
+
+_install_sonarrv3 () {
+    echo "Installing Sonarr v3 from apt" | tee -a $log
+    if [[ -z $sonarrv3owner ]];then
+        sonarrv3owner=$(cut -d: -f1 < /root/.master.info)
+    fi;
+    echo "Setting sonarr v3 owner to $sonarrv3owner" >> $log
+    # settings relevant from https://github.com/Sonarr/Sonarr/blob/phantom-develop/distribution/debian/config
+    echo "sonarr sonarr/owning_user  string ${sonarrv3owner}" | debconf-set-selections
+    echo "sonarr sonarr/owning_group string ${sonarrv3owner}" | debconf-set-selections
+    apt_install sonarr
+
+    if [[ ! -d /var/lib/sonarr ]]; then
+        echo "ERROR: The Sonarr v3 pacakge did not install correctly. Please try again. (Is sonarr repo reachable?)"
+        exit 1
+    fi
+}
+
+_add2usergroups_sonarrv3 () {
+        if [[ -z $sonarrv3grouplist ]]; then
+            if ask "Do you want to let Sonarr access other users' home directories?" N; then
+                echo "Space separated list of users to give sonarr access to: (e.g. \"user1 user2\")"
+                read -r sonarrv3grouplist
+            fi
+        fi
+
+        if [[ -n $sonarrv3grouplist ]]; then
+            for u in $sonarrv3grouplist; do
+                echo "Adding ${sonarrv3owner} to $u's group" | tee -a $log
+                usermod -a -G "$u" "$sonarrv3owner" | tee -a $log
+                chmod g+rwx /home/"$u"
+            done
+        fi
+}
+
+_nginx_sonarr () {
+    if [[ -f /install/.nginx.lock ]]; then
+        #TODO what is this sleep here for? See if this can be fixed by doing a check for whatever it needs to
+        sleep 20
```

Comment:
This was to make sure the config was properly populated before we attempt to manipulate it for nginx.

Sonarr can take a bit to start sometimes in my experience which is why I wanted to make sure it had enough time. If the config can just be `cat`'d in this can probably be ignored.

Reply:
`cat`'d in? Not sure I understand what that would achieve. Do you mean grepping for some content in a while loop?

Reply:
Oh I see what you mean. I'd rather let it populate itself as this might change over time and I don't want to maintain that, plus preserving an API key if any is gonna be a pain. We could instead use `xq` which is `jq` for xml though, and add that value in through that. then again, if sonarr does it, might as well wait

Reply:
Like if we are injecting the file to be pre-filled rather than waiting for it to be initialized then making changes with `sed`

Reply:
I'd leave this in seeing as the config needs to be present to set up the db and all from what I hear, will look at this if someone has an issue with it down the road